### ISSUE TITLE
"More" under facets may cause http request on https site causing mixed security mode and content blocking by browser

### ISSUE THREAD

**Issue by sfisher** [projectblacklight/blacklight#1487]
The module Blacklight::Controller#search_facet_url may create http links on an https site in some circumstances.  The AJAX request to load more content is then blocked by many browsers for carrying out an insecure on a secure site.  It may not be happening all the time, but happened in our use of Blacklight through Geoblacklight in our environment.

The simplest fix is to generate a path rather than a full URL that might mismatch the protocol of the original site by using the "only_path: true" option.

For example, changing:

`opts = search_state.to_h.merge(action: "facet").merge(options).except(:page)`

to 

`opts = search_state.to_h.merge(action: "facet", only_path: true).merge(options).except(:page)`

**Comment by cbeer** [comment]
That sounds reasonable -- although we'll want to rename the method `search_facet_path` if we do that.

I'm also curious what configuration causes this problem, we're running sites under HTTPS and haven't had any trouble like you describe.

**Comment by sfisher** [comment]
I'm not sure why it was causing us this problem, since I agree it seemed to work for many other people.  Our configuration is Apache (which redirects http pages to https) in front of standalone Passenger.  The blacklight is brought in through geoblacklight (1.1.2) which is also embedded in another engine for a portion of our application.

It's not an big problem for us since we've overridden the method in our own code to correct the issue we were having.  I just figured I might as well report what we ran into in case it comes up for other people.  I'm not sure why rails would create a http link in an https page.  Running Ruby 2.2.3 and Rails Rails 4.2.7.1.
